packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -186,4 +249,213 @@ export class AutoScalingGroup extends cdk.Construct implements ec2.IClassicLoadB
     public addToRolePolicy(statement: cdk.PolicyStatement) {
         this.role.addToPolicy(statement);
     }
+
+    /**
+     * Apply CloudFormation update policies for the AutoScalingGroup
+     */
+    private applyUpdatePolicies(props: AutoScalingGroupProps) {
+        if (props.updateType === UpdateType.ReplacingUpdate) {
+            this.asgUpdatePolicy.autoScalingReplacingUpdate = { willReplace: true };
+
+            if (props.replacingUpdateMinSuccessfulInstancesPercent !== undefined) {
+                // Yes, this goes on CreationPolicy, not as a process parameter to ReplacingUpdate.
+                // It's a little confusing, but the docs seem to explicitly state it will only be used
+                // during the update?
+                //
+                // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-creationpolicy.html
+                this.asgCreationPolicy.autoScalingCreationPolicy = {
```

Comment:
there is another creation policy that we use to prevent false positives when `userdata` must completely successfully:

```
  "ResourceSignal" : {    
    "Count" : Integer,
    "Timeout" : String
  }
```

Should we support this variation in this PR, or another PR?  I am actually running into this issue trying to port our bastion pattern that requires some `userdata` to associate an EIP for consistent A records.

Reply:
Yep, just added that to this PR as well:

https://github.com/awslabs/aws-cdk/pull/595/files/90118027119577fd40e5d80cbc4a2954f45ff873#diff-2db67fa7f5c77699be5c76d95371fcd5R279